internal/parameterizer/parameterizer_test.go

```diff
@@ -0,0 +1,247 @@
+
+	t.Run("6.IR with no services , check ingress parameterizer ", func(t *testing.T) {
+
+		// scMap should be 1, we can parameterize
+		ir := getIRWithoutServices()
+		irAfter, err := parameterize.Parameterize(ir)
+		if err != nil {
+			t.Fatal("Failed to parameterize. Error:", err)
+		}
+
+		hostParam := "{{ .Release.Name }}-"
+
+		if irAfter.TargetClusterSpec.Host != hostParam {
+			t.Fatal("Failed to parameterize")
+
+		}
+	})
+
+}
+
+func getIRWithoutServices() types.IR {
+	p := plantypes.NewPlan()
+	ir := types.NewIR(p)
+	return ir
+}
+
+func getIRWithServicesAndWithoutContainers() types.IR {
+	svcname1 := "svcname1"
+	svcname2 := "svcname2"
+	svc1 := types.Service{Name: svcname1, Replicas: 2}
+	svc2 := types.Service{Name: svcname2, Replicas: 2}
+	p := plantypes.NewPlan()
+	ir := types.NewIR(p)
+	ir.Services[svcname1] = svc1
+	ir.Services[svcname2] = svc2
+	return ir
+}
+
+func getIRWithServicesAndContainers() types.IR {
+	c1 := corev1.Container{
+		Name: "container-1",
+	}
+	c2 := corev1.Container{
+		Name: "container-2",
+	}
+	//c1.ImagePullPolicy = v1.PullAlways // not needed for the moment
+	//c2.ImagePullPolicy = v1.PullAlways // not needed for the moment
+	svcname1 := "svcname1"
+	svcname2 := "svcname2"
+	svc1 := types.Service{Name: svcname1, Replicas: 2}
+	svc2 := types.Service{Name: svcname2, Replicas: 4}
+	svc1.Containers = append(svc1.Containers, c1)
+	svc2.Containers = append(svc2.Containers, c2)
+	p := plantypes.NewPlan()
+	ir := types.NewIR(p)
+	ir.Services[svcname1] = svc1
+	ir.Services[svcname2] = svc2
+	return ir
+}
+
+func getIRWithStoragePVCKind() types.IR {
+
+	storage1cn := "storage-1cn" // example class name
+	storage1 := irtypes.Storage{
+		StorageType: irtypes.PVCKind,
+		Name:        "storage-1",
+		PersistentVolumeClaimSpec: corev1.PersistentVolumeClaimSpec{
+			VolumeName: "storage-1",
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: common.DefaultPVCSize,
+				},
+			},
+			StorageClassName: &storage1cn,
+		}}
+
+	p := plantypes.NewPlan()
+	ir := types.NewIR(p)
+
+	ir.Storages = append(ir.Storages, storage1)
+	return ir
+}
+
+func getIRWithStorageNotPVCKind() types.IR {
+	// instead of using a storage type PVCKind, we use SecretKind
+
+	storage1cn := "storage-1cn" // example class name
+
+	storage1 := irtypes.Storage{
+		StorageType: irtypes.SecretKind,
+		Name:        "storage-1",
+		PersistentVolumeClaimSpec: corev1.PersistentVolumeClaimSpec{
+			VolumeName: "storage-1",
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: common.DefaultPVCSize,
+				},
+			},
+			StorageClassName: &storage1cn,
+		}}
+
+	p := plantypes.NewPlan()
+	ir := types.NewIR(p)
+	ir.Storages = append(ir.Storages, storage1)
+	return ir
+}
+
+func getIRWithContainer() types.IR {
+
+	// Setup Containers
+	cname1 := "name1"
+	cnew1 := true
+	cont1 := types.NewContainer(plantypes.DockerFileContainerBuildTypeValue,
+		cname1,
```

Comment:
I think there's a formatting problem here since it isn't lined up.
Is `go fmt` being run when you save? Usually vscode will do that, if not please configure it to do so.